pkg/kclient/jobs.go

```diff
@@ -0,0 +1,103 @@
+package kclient
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/klog"
+)
+
+// constants for volumes
+const (
+	JobsKind          = "Job"
+	JobsResource      = "jobs"
+	JobsAPIVersion    = "batch/v1"
+	executeJobTimeout = 1 * time.Minute
+)
+
+// CreateJobs creates a K8s job to execute task
+func (c *Client) CreateJobs(job batchv1.Job, namespace string) (*batchv1.Job, error) {
+	if namespace == "" {
+		namespace = c.Namespace
+	}
+	createdJob, err := c.KubeClient.BatchV1().Jobs(namespace).Create(context.TODO(), &job, metav1.CreateOptions{FieldManager: FieldManager})
```

Comment:
Another point I think we discussed last week. This call might not succeed if you re-run the deploy command before the previous job gets deleted:

```
$ odo deploy
  __
 /  \__     Running the application in Deploy mode using my-dotnet-exec-deploy Devfile
 \__/  \    Namespace: default
 /  \__/    odo version: v3.6.0
 \__/

↪ Executing command in container (command: deploy)
 ✗  Executing "ls -lhrt /" [5ms]
 ✗  unable to create Jobs: jobs.batch "my-dotnet-exec-deploy-app-deploy" already exists
```

Reply:
This has been addressed in 3 ways: 
- Delete any related preexisting jobs before creating a new one; this helps in cleaning any residual failed job of an abruptly terminated `odo deploy`
- Delete a created job before exiting - this works if `odo deploy`  terminated naturally
- As a backup, set `TTLSecondsAfterFinished: 60s` - this helps in deleting any job that might have completed after user abruptly terminated `odo deploy`.

I hope this helps. Thank you for this suggestion!